### PR TITLE
[New @deck.gl/json syntax support - Part 3] @deck.gl/json: Add function and enum identification syntax

### PR DESCRIPTION
**Pull request by ajduberstein**
#### Background
Incorporates type hints for enums and constants into @deck.gl/json.

Related to [the syntactic sugar RFC](https://github.com/uber/deck.gl/blob/92b047fa3b6040c6d2810fe280eaf6bfad8fd70f/dev-docs/RFCs/vNext/json-converter-keyword-handling-rfc.md).

#### Change List
- Provide hint to parse enums and constants

### CODE REVIEW

File: modules/json/src/json-converter.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ import {instantiateClass} from './helpers/instantiate-class';
 import parseJSON from './helpers/parse-json';
 
 const isObject = value => value && typeof value === 'object';
+const FUNCTION_IDENTIFIER = '@@=';
+const ENUM_IDENTIFIER = '@@#';
+const CONSTANT_IDENTIFIER = '@@!';
```
Comment (Pessimistress) [inline modules/json/src/json-converter.js]: What is constant vs. enum?

File: test/modules/json/data/deck-props.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "latitude": 37.8,
     "zoom": 12
   },
-  "controller": "MapController",
+  "controller": "@@!MapController",
```
Comment (Pessimistress) [inline test/modules/json/data/deck-props.json]: Why is this needed? `controller: true` would work with any type of view.

File: modules/json/src/json-converter.js (hunk 1 above)
Comment (ibgreen) [reply modules/json/src/json-converter.js]: Yes, I am not sure we need to identifiers / namespaces. 

If we add a required prefix to force evaluation of a string as a constant, both function-valued constants (like `MapController`, `LinearInterpolator`, ...) and integer/string valued enums (like `COORDINATE_SYSTEM.METER_OFFSETS`) should be able to go into the same string-to-value map in the JSON converter. 

The `JSONConverter` may need a little TLC in this area, the constant handling was done rather quickly.

File: test/modules/json/data/deck-props.json (hunk 2 above)
Comment (ajduberstein) [reply test/modules/json/data/deck-props.json]: This is just a fixture, used to test the use of the `constants` lookup configured in `test/modules/json/json-configuration-for-deck.js` on line 17.

Comment (ajduberstein) [reply test/modules/json/data/deck-props.json]: I'll find a more practical example, since the user would never modify this directly
